src/plugins/vis_type_timeseries/public/application/visualizations/views/timeseries/utils/series_domain_calculation.ts

```diff
@@ -0,0 +1,25 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import { PanelData } from '../../../../../../common/types';
+
+interface DomainBounds {
+  domainStart: number;
+  domainEnd: number;
+}
+
+export const calculateDomainForSeries = (series: PanelData[]): DomainBounds => {
+  const seriesData = series[0]?.data || [];
+
+  return seriesData?.length
+    ? {
+        domainStart: seriesData[0][0],
+        domainEnd: seriesData[Math.max(seriesData.length - 1, 0)][0],
+      }
+    : ({} as DomainBounds);
```

Comment:
please avoid of using casting like `{} as DomainBounds`. Devs who will use your method expect getting `DomainBounds` object for all cases. In fact you return `{}`

I see 3 options:
1.
`(series: PanelData[]): DomainBounds | {}`
2. 
```
 interface DomainBounds {
  domainStart?: number;
  domainEnd?: number;
}
```
3. return` null/undefined.` - I like it more